src/context/authStore.test.tsx

```diff
@@ -0,0 +1,284 @@
+              {JSON.stringify(authCredentials[testCampaignKey])}
+            </Text>
+          )}
+        </AuthStoreContext.Consumer>
+      </AuthStoreContextProvider>
+    );
+
+    expect(mockGetItem).toHaveBeenCalledTimes(1);
+    expect(mockGetItem).toHaveBeenCalledWith("AUTH_STORE");
+
+    await wait(() => {
+      expect(queryByTestId("credentials")).toHaveTextContent("");
+    });
+  });
+
+  it("should call Sentry when the campaign config from the store is malformed", async () => {
+    expect.assertions(3);
+    mockGetItem.mockImplementationOnce(() => "malformed object");
+
+    render(
+      <ErrorBoundary>
+        <AuthStoreContextProvider shouldMigrate={false}>
+          <AuthStoreContext.Consumer>
+            {({ authCredentials }) => (
+              <Text testID="credentials">
+                {JSON.stringify(authCredentials[testCampaignKey])}
+              </Text>
+            )}
+          </AuthStoreContext.Consumer>
+        </AuthStoreContextProvider>
+      </ErrorBoundary>
+    );
+
+    expect(mockGetItem).toHaveBeenCalledTimes(1);
+    expect(mockGetItem).toHaveBeenCalledWith("AUTH_STORE");
+
+    await wait(() => {
+      expect(mockCaptureException).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  it("should clear the auth credentials and from asyncstorage when clear function is called", async () => {
+    expect.assertions(6);
+    mockGetItem.mockImplementationOnce(() =>
+      JSON.stringify({
+        [testCampaignKey]: {
+          operatorToken: "operatorToken",
+          sessionToken: "sessionToken",
+          endpoint: "endpoint",
+          expiry: 0
+        }
+      })
+    );
+
+    const { queryByTestId, getByText } = render(
+      <AuthStoreContextProvider shouldMigrate={false}>
+        <AuthStoreContext.Consumer>
+          {({ authCredentials, clearAuthCredentials }) => (
+            <>
+              <Text testID="credentials">
+                {JSON.stringify(authCredentials[testCampaignKey])}
+              </Text>
+              <Button
+                onPress={() => clearAuthCredentials()}
+                title="test button"
+              />
+            </>
+          )}
+        </AuthStoreContext.Consumer>
+      </AuthStoreContextProvider>
+    );
+
+    expect(mockGetItem).toHaveBeenCalledTimes(1);
+    expect(mockGetItem).toHaveBeenCalledWith("AUTH_STORE");
+
+    await wait(() => {
+      expect(queryByTestId("credentials")).toHaveTextContent(
+        `{"operatorToken":"operatorToken","sessionToken":"sessionToken","endpoint":"endpoint","expiry":0}`
+      );
+    });
+
+    const button = getByText("test button");
+    fireEvent.press(button);
+    await wait(() => {
+      expect(mockSetItem).toHaveBeenCalledTimes(1);
+      expect(mockSetItem).toHaveBeenCalledWith("AUTH_STORE", "{}");
+      expect(queryByTestId("credentials")).toHaveTextContent("");
+    });
+  });
+
+  it("should add a set of auth credentials properly", async () => {
```

Comment:
for this test, we're testing the case where its a new set of auth credentials being added. Do you think we should have a case to show that we append a new set of auth credentials to an existing set?

I'm assuming authStore works the same way as campaignConfigStore haha, so all the auths under 1 key, unless I'm wrong haha

Reply:
Have just added the tests!